web_ui/src/providers/label-validation-provider/label-validation-provider.component.tsx

```diff
@@ -0,0 +1,26 @@
+// Copyright (C) 2022-2025 Intel Corporation
+// LIMITED EDGE SOFTWARE DISTRIBUTION LICENSE
+
+import { createContext, ReactNode, useContext } from 'react';
+
+import { MissingProviderError } from '../../shared/missing-provider-error';
+import { UseLabelValidationProps, UseLabelValidationResult } from './label-validation-types';
+import { useLabelValidation } from './use-label-validation.hook';
+
+const LabelValidationContext = createContext<UseLabelValidationResult | null>(null);
+
+export const LabelValidationProvider = ({ children, ...props }: UseLabelValidationProps & { children: ReactNode }) => {
+    const validation = useLabelValidation(props);
+
+    return <LabelValidationContext.Provider value={validation}>{children}</LabelValidationContext.Provider>;
+};
```

Comment:
I don't really see the use case of this provider.
As a user of the provider, why wouldn't I just do `useLabelValidation(props)` instead of going through the provider?

Reply:
im not sure if just the hook will do since this is shared logic between all label components on the tree. But once i start replacing it i will be able to confirm this

Reply:
I removed the provider for now. Im not entirely sure if we're gonna need to share state. But for now ill start with the simple/naive approach and handle each label input separately.